src/openrct2/world/Map.cpp

```diff
@@ -1500,12 +1522,19 @@ void map_update_tiles()
             interleaved_xy >>= 1;
         }
 
-        auto mapPos = TileCoordsXY{ x, y }.ToCoordsXY();
-        auto* surfaceElement = map_get_surface_element_at(mapPos);
-        if (surfaceElement != nullptr)
+        // Repeat for each 256x256 block on the map
+        for (int32_t blockY = 0; blockY < gMapSize; blockY += 256)
         {
-            surfaceElement->UpdateGrassLength(mapPos);
-            scenery_update_tile(mapPos);
+            for (int32_t blockX = 0; blockX < gMapSize; blockX += 256)
```

Comment:
This looks wrong. BlockX seems to contain big coords - you check against gMapSize and add 256 on every iteration -, yet the value is used to create a TIleCoordsXY.

Reply:
it isn't used until the nsf increases the map size. In normal map size it only ever does the one iteration.